docs/examples/water-value.ipynb

```diff
@@ -0,0 +1,541 @@
+   "outputs": [],
+   "source": [
+    "n.buses_t.marginal_price[\"electricity\"].resample(\"D\").mean().plot(\n",
+    "    figsize=(6, 2), ylabel=\"€/MWh\", xlabel=\"\"\n",
+    ");"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a2e39ce6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pdc = (\n",
+    "    n.buses_t.marginal_price[\"electricity\"]\n",
+    "    .sort_values(ascending=False)\n",
+    "    .reset_index(drop=True)\n",
+    ")\n",
+    "pdc.plot(figsize=(5, 3), ylabel=\"€/MWh\", xlabel=\"snapshots\", xlim=(0, 8760 / 3));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "87868e78",
+   "metadata": {},
+   "source": [
+    "The price at the hydrogen bus reflects the marginal storage value (MSV) of hydrogen. The MSV indicates the willingness to pay for an additional unit of hydrogen stored at that time and, thereby, reflects its future value. This is the equivalent of water values in hydro storage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f4cb6e05",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "msv = n.buses_t.marginal_price[\"hydrogen\"]\n",
+    "msv.plot(figsize=(6, 2), ylim=(0, 100), ylabel=\"€/MWh\", xlabel=\"\");"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ac44174b",
+   "metadata": {},
+   "source": [
+    "As in this case with perfect foresight, the model foresees future scarcity periods in electricity supply, the storage is operated seasonally, retaining high storage levels over the summer to cover scarcity periods in winter. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "821af3ed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "soc = n.stores_t.e.div(1e3).squeeze()\n",
+    "soc.plot(figsize=(6, 2), ylabel=\"GWh\", xlabel=\"\", legend=False);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "177be6be",
+   "metadata": {},
+   "source": [
+    "## Case 2: Myopic Foresight\n",
+    "\n",
+    "Now, we solve the same dispatch problem, but with myopic foresight of 15 days (120 snapshots at 3-hourly resolution). This means that at each optimisation step, the model only considers the next 15 days when making operational decisions, without knowledge of future periods beyond this horizon."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "44a14c8f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n.model.solver_model = None\n",
+    "n2 = n.copy()\n",
+    "\n",
+    "n2.optimize.optimize_with_rolling_horizon(\n",
+    "    assign_all_duals=True,\n",
+    "    horizon=120,\n",
+    "    overlap=0,\n",
+    "    log_to_console=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14dfc7d5",
+   "metadata": {},
+   "source": [
+    "This leads to higher fuel costs of 1337 M€/a, as the storage is not optimally utilised for seasonal shifting and the gas generator must step in."
```

Comment:
I wouldn't put the actual number into the markdown comment (same in an earlier comment) as it can easily be missed if something changes in the example network and updates are made to the examples